lib/cli/src/utils.rs

```diff
@@ -40,14 +40,53 @@ pub fn parse_mapdir(entry: &str) -> Result<(String, PathBuf)> {
     }
 }
 
-/// Parses a mapdir from an env var
+/// Parses an environment variable.
 pub fn parse_envvar(entry: &str) -> Result<(String, String)> {
-    if let [env_var, value] = entry.split('=').collect::<Vec<&str>>()[..] {
-        Ok((env_var.to_string(), value.to_string()))
-    } else {
-        bail!(
-            "Env vars must be of the form <var_name>=<value>. Found {}",
+    let entry = entry.trim();
+
+    match entry.find('=') {
+        None => bail!(
+            "Environment variable must be of the form `<name>=<value>`; found `{}`",
+            &entry
+        ),
+
+        Some(0) => bail!(
+            "Environment variable is not well formed, the `name` is missing in `<name>=<value>`; got `{}`",
+            &entry
+        ),
+
+        Some(position) if position == entry.len() - 1 => bail!(
+            "Environment variable is not well formed, the `value` is missing in `<name>=<value>`; got `{}`",
             &entry
+        ),
+
+        Some(position) => Ok((entry[..position].into(), entry[position + 1..].into())),
```

Comment:
The actual splitting can just be done with just https://doc.rust-lang.org/std/primitive.str.html#method.split_at ; but we won't have as many error cases to report.

Just a heads up, I'm fine keeping it as-is

Reply:
We would need to deal with empty string for the left- or right-end side of the `=`. I think the code would not be simplified that much. Thanks for the heads-up!